<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from human.urdf.xacro               | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="human">
  <material name="color">
    <color rgba="1 0.86 0.67 1"/>
  </material>
  <material name="gray">
    <color rgba="0.42 0.4 0.42 0.4"/>
  </material>
  <material name="darkgray">
    <color rgba="0.3 0.3 0.3 0.5"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 0.0"/>
  </material>
  <!--To open this model with Gazebo replace the null masses, inertias and dimensions of the 'fake links f1 and f2'-->
  <!--LINKS-->
  <!--Link base (1)-->
  <link name="Hip_mocap">
    <inertial>
      <mass value="5.696"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="hip">
    <inertial>
      <mass value="5.696"/>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.02" iyz="0" izz="0.06"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <box size="0.08 0.16 0.08"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <box size="0.08 0.16 0.08"/>
      </geometry>
    </collision>
  </link>
  <link name="Ab_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Ab_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Ab">
    <inertial>
      <mass value="7.2624"/>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <inertia ixx="0.05" ixy="0" ixz="0" iyy="0.04" iyz="0" izz="0.08"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <box size="0.08 0.16 0.08"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <box size="0.08 0.16 0.08"/>
      </geometry>
    </collision>
  </link>
  <link name="Chest_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Chest_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Chest">
    <inertial>
      <mass value="2.848"/>
      <origin rpy="0 0 0" xyz="0 0 0.075"/>
      <inertia ixx="0.0040381" ixy="0" ixz="0" iyy="0.0077541" iyz="0" izz="0.0053923"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.075"/>
      <geometry>
        <box size="0.08 0.16 0.15"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.075"/>
      <geometry>
        <box size="0.08 0.16 0.12"/>
      </geometry>
    </collision>
  </link>
  <link name="Neck_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Neck_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Neck">
    <inertial>
      <mass value="0.8544"/>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <inertia ixx="0.00086442" ixy="0" ixz="0" iyy="0.00086442" iyz="0" izz="0.00040681"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <cylinder length="0.08" radius="0.01543"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.04"/>
      <geometry>
        <cylinder length="0.08" radius="0.01543"/>
      </geometry>
    </collision>
  </link>
  <link name="Head_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Head_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="Head">
    <inertial>
      <mass value="2.5632"/>
      <origin rpy="0 0 0" xyz="0           0    0.099452"/>
      <inertia ixx="0.010141" ixy="0" ixz="0" iyy="0.010141" iyz="0" izz="0.010141"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0           0    0.099452"/>
      <geometry>
        <sphere radius="0.099452"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0           0    0.099452"/>
      <geometry>
        <sphere radius="0.099452"/>
      </geometry>
    </collision>
  </link>
  <!--Right Arm-->
  <link name="RightShoulder_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightShoulder_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightShoulder">
    <inertial>
      <mass value="2.2072"/>
      <origin rpy="1.5708 0 0" xyz="0  -0.075 0"/>
      <inertia ixx="0.0050082" ixy="0" ixz="0" iyy="0.00086686" iyz="0" izz="0.0050082"/>
    </inertial>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0  -0.075 0"/>
      <geometry>
        <cylinder length="0.15" radius="0.028026"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0  -0.075 0"/>
      <geometry>
        <cylinder length="0.15" radius="0.028026"/>
      </geometry>
    </collision>
  </link>
  <link name="RightUpperArm_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightUpperArm_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightUpperArm">
    <inertial>
      <mass value="2.136"/>
      <origin rpy="1.5708 0 0" xyz="0    -0.14573           0"/>
      <inertia ixx="0.015663" ixy="0" ixz="0" iyy="0.0010831" iyz="0" izz="0.015663"/>
    </inertial>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0    -0.14573           0"/>
      <geometry>
        <cylinder length="0.29147" radius="0.031846"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0    -0.14573           0"/>
      <geometry>
        <cylinder length="0.29147" radius="0.031846"/>
      </geometry>
    </collision>
  </link>
  <link name="RightForeArm_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightForeArm_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightForeArm">
    <inertial>
      <mass value="1.424"/>
      <origin rpy="1.5708 0 0" xyz="0    -0.12047           0"/>
      <inertia ixx="0.0070493" ixy="0" ixz="0" iyy="0.00032093" iyz="0" izz="0.0070493"/>
    </inertial>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0    -0.12047           0"/>
      <geometry>
        <cylinder length="0.24094" radius="0.021231"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0    -0.12047           0"/>
      <geometry>
        <cylinder length="0.24094" radius="0.021231"/>
      </geometry>
    </collision>
  </link>
  <link name="RightHand_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightHand_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="RightHand">
    <inertial>
      <mass value="0.4272"/>
      <origin rpy="0 0 0" xyz="0 -0.075 0"/>
      <inertia ixx="0.0012426" ixy="0" ixz="0" iyy="0.00058792" iyz="0" izz="0.0017021"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 -0.075 0"/>
      <geometry>
        <box size="0.12 0.15 0.035"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 -0.075 0"/>
      <geometry>
        <box size="0.12 0.15 0.035"/>
      </geometry>
    </collision>
  </link>
  <link name="RightHandCOM">
    <inertial>
      <mass value="0.00001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0   0           0"/>
      <geometry>
        <box size="0     0    0"/>
      </geometry>
      <material name="color"/>
    </visual>
  </link>
  <!--Chain from (12) to (15)-->
  <link name="LeftShoulder_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftShoulder_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftShoulder">
    <inertial>
      <mass value="2.2072"/>
      <!--COM origin wrt jLeftC7Shoulder-->
      <origin rpy="1.5708 0 0" xyz="0  0.075 0"/>
      <inertia ixx="0.0050082" ixy="0" ixz="0" iyy="0.00086686" iyz="0" izz="0.0050082"/>
    </inertial>
    <visual>
      <!--box origin wrt jLeftC7Shoulder. RPY rotated of pi/2.-->
      <origin rpy="1.5708 0 0" xyz="0  0.075 0"/>
      <geometry>
        <cylinder length="0.15" radius="0.028026"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0  0.075 0"/>
      <geometry>
        <cylinder length="0.15" radius="0.028026"/>
      </geometry>
    </collision>
  </link>
  <link name="LeftUpperArm_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftUpperArm_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftUpperArm">
    <inertial>
      <mass value="2.136"/>
      <origin rpy="1.5708 0 0" xyz="0     0.14573           0"/>
      <inertia ixx="0.015663" ixy="0" ixz="0" iyy="0.0010831" iyz="0" izz="0.015663"/>
    </inertial>
    <visual>
      <origin rpy="1.5708 0 0" xyz="0     0.14573           0"/>
      <geometry>
        <cylinder length="0.29147" radius="0.031846"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0     0.14573           0"/>
      <geometry>
        <cylinder length="0.29147" radius="0.031846"/>
      </geometry>
    </collision>
  </link>
  <link name="LeftForeArm_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftForeArm_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftForeArm">
    <inertial>
      <mass value="1.424"/>
      <!--COM origin wrt jLeftElbow-->
      <origin rpy="1.5708 0 0" xyz="0     0.12047           0"/>
      <inertia ixx="0.0070493" ixy="0" ixz="0" iyy="0.00032093" iyz="0" izz="0.0070493"/>
    </inertial>
    <visual>
      <!--box origin wrt jLeftElbow. RPY rotated of pi/2.-->
      <origin rpy="1.5708 0 0" xyz="0     0.12047           0"/>
      <geometry>
        <cylinder length="0.24094" radius="0.021231"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <!--box origin wrt jLeftElbow. RPY rotated of pi/2.-->
      <origin rpy="1.5708 0 0" xyz="0     0.12047           0"/>
      <geometry>
        <cylinder length="0.24094" radius="0.021231"/>
      </geometry>
    </collision>
  </link>
  <link name="LeftHand_f1">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftHand_f2">
    <inertial>
      <mass value="0"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
  </link>
  <link name="LeftHand">
    <inertial>
      <mass value="0.4272"/>
      <origin rpy="0 0 0" xyz="0 0.075 0"/>
      <inertia ixx="0.0012426" ixy="0" ixz="0" iyy="0.00058792" iyz="0" izz="0.0017021"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0.075 0"/>
      <geometry>
        <box size="0.12 0.15 0.035"/>
      </geometry>
      <material name="color"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0.075 0"/>
      <geometry>
        <box size="0.12 0.15 0.035"/>
      </geometry>
    </collision>
  </link>
  <link name="LeftHandCOM">
    <inertial>
      <mass value="0.00001"/>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0    0           0"/>
      <geometry>
        <box size="0     0    0"/>
      </geometry>
      <material name="color"/>
    </visual>
  </link>
  <!--JOINTS-->
  <joint name="static_offset_mocap" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Hip_mocap"/>
    <child link="hip"/>
    <axis xyz="0 0 0"/>
  </joint>
  <joint name="Ab_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.08"/>
    <parent link="hip"/>
    <child link="Ab_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.610865" upper="0.610865" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="Ab_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Ab_f1"/>
    <child link="Ab_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.523599" upper="1.309" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="Ab_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Ab_f2"/>
    <child link="Ab"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.610865" upper="0.610865" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="Chest_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.08"/>
    <parent link="Ab"/>
    <child link="Chest_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.523599" upper="1.309" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="Chest_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Chest_f1"/>
    <child link="Chest_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.349066" upper="0.349066" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="Chest_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Chest_f2"/>
    <child link="Chest"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.261799" upper="0.698132" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="Neck_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.15"/>
    <parent link="Chest"/>
    <child link="Neck_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.610865" upper="0.610865" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="Neck_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Neck_f1"/>
    <child link="Neck_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.959931" upper="1.5708" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="Neck_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Neck_f2"/>
    <child link="Neck"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.22173" upper="1.22173" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="Head_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0.08"/>
    <parent link="Neck"/>
    <child link="Head_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.610865" upper="0.610865" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="Head_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Head_f1"/>
    <child link="Head_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.436332" upper="0.174533" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="Head_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="Head_f2"/>
    <child link="Head"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.436332" upper="0.174533" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <!--Right Arm-->
  <joint name="jRightC7Shoulder_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0   -0.029713 0.15"/>
    <parent link="Chest"/>
    <child link="RightShoulder_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.785398" upper="0.0872665" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jRightC7Shoulder_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightShoulder_f1"/>
    <child link="RightShoulder_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.785398" upper="0.0872665" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jRightC7Shoulder_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightShoulder_f2"/>
    <child link="RightShoulder"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.785398" upper="0.0872665" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jRightShoulder_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0  -0.15  0"/>
    <parent link="RightShoulder"/>
    <child link="RightUpperArm_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-2.35619" upper="1.5708" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jRightShoulder_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightUpperArm_f1"/>
    <child link="RightUpperArm_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.5708" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jRightShoulder_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightUpperArm_f2"/>
    <child link="RightUpperArm"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.785398" upper="3.14159" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jRightElbow_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 -0.3 0"/>
    <parent link="RightUpperArm"/>
    <child link="RightForeArm_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="0" upper="2.53073" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jRightElbow_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightForeArm_f1"/>
    <child link="RightForeArm_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.48353" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jRightElbow_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightForeArm_f2"/>
    <child link="RightForeArm"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.48353" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jRightWrist_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0.000102 -0.24 0"/>
    <parent link="RightForeArm"/>
    <child link="RightHand_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.872665" upper="1.0472" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jRightWrist_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightHand_f1"/>
    <child link="RightHand_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.872665" upper="1.0472" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jRightWrist_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="RightHand_f2"/>
    <child link="RightHand"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.523599" upper="0.349066" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jRightHandCOM" type="fixed">
    <origin rpy="1.57 0 0" xyz="0 -0.075 0"/>
    <parent link="RightHand"/>
    <child link="RightHandCOM"/>
    <axis xyz="0 0 0"/>
  </joint>
  <!--Left Arm-->
  <joint name="jLeftC7Shoulder_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.029713 0.15"/>
    <parent link="Chest"/>
    <child link="LeftShoulder_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.0872665" upper="0.785398" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jLeftC7Shoulder_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftShoulder_f1"/>
    <child link="LeftShoulder_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.0872665" upper="0.785398" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jLeftC7Shoulder_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftShoulder_f2"/>
    <child link="LeftShoulder"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.0872665" upper="0.785398" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jLeftShoulder_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.15 0"/>
    <parent link="LeftShoulder"/>
    <child link="LeftUpperArm_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="2.35619" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jLeftShoulder_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftUpperArm_f1"/>
    <child link="LeftUpperArm_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.5708" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jLeftShoulder_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftUpperArm_f2"/>
    <child link="LeftUpperArm"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-3.14159" upper="0.785398" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jLeftElbow_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0 0.3 0"/>
    <parent link="LeftUpperArm"/>
    <child link="LeftForeArm_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.48353" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jLeftElbow_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftForeArm_f1"/>
    <child link="LeftForeArm_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.5708" upper="1.48353" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jLeftElbow_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftForeArm_f2"/>
    <child link="LeftForeArm"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-2.53073" upper="0" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jLeftWrist_rotx" type="revolute">
    <origin rpy="0 0 0" xyz="0.000102 0.24 0"/>
    <parent link="LeftForeArm"/>
    <child link="LeftHand_f1"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-1.0472" upper="0.872665" velocity="1.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="jLeftWrist_roty" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftHand_f1"/>
    <child link="LeftHand_f2"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.349066" upper="0.523599" velocity="1.0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="jLeftWrist_rotz" type="revolute">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="LeftHand_f2"/>
    <child link="LeftHand"/>
    <dynamics damping="0.1" friction="0.0"/>
    <limit effort="30" lower="-0.349066" upper="0.523599" velocity="1.0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="jLeftHandCOM" type="fixed">
    <origin rpy="-1.57 0 0" xyz="0 0.075 0"/>
    <parent link="LeftHand"/>
    <child link="LeftHandCOM"/>
    <axis xyz="0 0 0"/>
  </joint>
</robot>
